packages/bpk-component-button/src/BpkButton-test.js

```diff
@@ -17,10 +17,25 @@
  */
 
 import React from 'react';
+import { axe } from 'jest-axe';
 import renderer from 'react-test-renderer';
+import { render } from '@testing-library/react';
 
 import BpkButton from './BpkButton';
 
+const defaultTestProps = {
+  onClick: () => {},
+  children: 'My button',
+};
+
+describe('BpkButton accessibility tests', () => {
+  it('should not have programmatically detectable accessibility issues', async () => {
+    const { container } = render(<BpkButton {...defaultTestProps} />);
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+});
+
```

Comment:
Are we able to reuse code in existing snapshot tests or examples instead of having a 3rd place that consumes the components?

Reply:
@georgegillams could you expand on this comment? Looking at this test it looks the exact same as how we would make any of the others below. Would be interested to chat to understand this comment more

Reply:
As things stand, we need to create an example usage of BpkButton for `examples.js`. Then we also need to create it in `BpkButton-test.js` to test how it's rendered, and again to test is for accessibility.

Maybe instead we could define an example, and then reuse that example for both snapshot and a11y testing?

Reply:
Yeah that's definitely an option. We could do this:

```js
it('should render correctly', () => {
  const { asFragment, container } = render(<BpkButton>My button</BpkButton>);
  expect(asFragment()).toMatchSnapshot();
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
```

The reason I didn't go for this was for two reasons:

1. It feels a little like it breaks single responsibility principle?
2. Banana doesn't do this and I'm using them as the gold standard of automated a11y testing

Reply:
Actually, let's ask Dom for his opinion. I'll loop him in.

Reply:
With regards to sharing `examples.js` in tests, I think that's out of scope for this. If we choose to do that it can be done separately without affecting this work.

Reply:
I would always have the a11y test as a separate test.

However, it can be within the same describe block (just a new `it()`), and doesn't need to use the `{...getDefaultTestProps()}` pattern if the test file doesn't use it. That pattern is one we use only if there are large amount of props we want to have test defaults shared across tests, but in this case the two props are also easy to declare just within the individual test.